app/models/manageiq/providers/azure/manager_mixin.rb

```diff
@@ -10,6 +10,7 @@ def connect(options = {})
   end
 
   def verify_credentials(_auth_type = nil, options = {})
+    require 'azure-armrest'
```

Comment:
@bdunne confirmed the `Azure::Armrest::UnauthorizedException` is loaded by requiring the library. 👍